File: 0225 implement-stack-using-queues/solution.java
class MyStack {
   private ArrayList<Integer> stack;
    private int top;

    public MyStack() {
        stack = new ArrayList<>();
    }
    
    public void push(int x) {
        stack.add(x);
    }
    
    public int pop() {
        if(stack.isEmpty()) return -1;
        return stack.remove(stack.size()-1);
    }
    
    public int top() {
        if(stack.isEmpty()) return -1;
        return stack.get(stack.size()-1);
    }
    
    public boolean empty() {
        return stack.isEmpty();
    }
}

/**
 * Your MyStack object will be instantiated and called as such:
 * MyStack obj = new MyStack();
 * obj.push(x);
 * int param_2 = obj.pop();
 * int param_3 = obj.top();
 * boolean param_4 = obj.empty();
 */

File: 0225 implement-stack-using-queues/solution.go
type MyStack struct {
    q []int
}


func Constructor() MyStack {
   return MyStack{
        q: []int{},
    }
}


func (this *MyStack) Push(x int)  {
    this.q = append(this.q,x)
    n:=len(this.q)
    for i:=0;i<n-1;i++ {
        t:=this.q[0]
        this.q = this.q[1:]
        this.q = append(this.q,t)
    }
}


func (this *MyStack) Pop() int {
    val := this.q[0]
    this.q = this.q[1:]
    return val
}


func (this *MyStack) Top() int { 
    return this.q[0]
}


func (this *MyStack) Empty() bool {
    return len(this.q) == 0
}


/**
 * Your MyStack object will be instantiated and called as such:
 * obj := Constructor();
 * obj.Push(x);
 * param_2 := obj.Pop();
 * param_3 := obj.Top();
 * param_4 := obj.Empty();
 */
